### Requirements:
- [registered discord bot][1]
- Administrator perms

**note:** to quickly see the bot in motion, skip to [the fifth cell](#cell-5)

[1]: https://discord.com/developers/applications

# Getting started
First step, download the code 😎

In [ ]:
!git clone https://github.com/0x6F72656F73/ImaginaryTicketing

Cloning into 'ImaginaryTicketing'...
remote: Enumerating objects: 57, done.
remote: Counting objects: 100% (57/57), done.
remote: Compressing objects: 100% (38/38), done.
remote: Total 57 (delta 18), reused 50 (delta 13), pack-reused 0
Unpacking objects: 100% (57/57), done.


In [ ]:
%cd ImaginaryTicketing/

/content/ImaginaryTicketing


# Installing requirements
Now install the requirements for the bot with pip


In [ ]:
!pip install -r requirements.txt 

     |████████████████████████████████| 3.8MB 9.7MB/s 
     |████████████████████████████████| 133kB 43.1MB/s 
     |████████████████████████████████| 788kB 37.7MB/s 
     |████████████████████████████████| 215kB 47.5MB/s 
     |████████████████████████████████| 1.3MB 33.0MB/s 
     |████████████████████████████████| 143kB 41.9MB/s 
     |████████████████████████████████| 296kB 41.9MB/s 
  Created wheel for grapheme: filename=grapheme-0.6.0-cp37-none-any.whl size=210097 sha256=cb26a8b5bf15c3e33394d6b05589aaff2e97e1e9a6c37709ac6ce13af1ad4c46
  Stored in directory: /root/.cache/pip/wheels/1e/f7/2c/341b06952dda321b8550d9ebd573dff4c0640fa710d9f23ecd
Successfully built grapheme


In [ ]:
%cd src/

/content/ImaginaryTicketing/src


Now put your bot token and prefix in a .env file.
<a name="cell-5"></a>


In [ ]:
!echo "DISCORD_TOKEN=ODM5NjM0MTAxNDgwMzI1MjEw.YJMgMQ.rx8XqmlnOjNUfJ3DaXykCoa1fM8" > .env;
!echo "BOT_PREFIX = ['$']" >> .env; cat .env

DISCORD_TOKEN=ODM5NjM0MTAxNDgwMzI1MjEw.YJMgMQ.rx8XqmlnOjNUfJ3DaXykCoa1fM8
BOT_PREFIX = ['t.']


The last step is getting the database to work. There is some logic in the bot that requires sqlite 3.24 and above. In my experience, simply doing `apt install sqlite3` doesn't work.

In [ ]:
!apt install sqlite3

Reading package lists... Done
Building dependency tree       
Reading state information... Done
Suggested packages:
  sqlite3-doc
The following NEW packages will be installed:
  sqlite3
0 upgraded, 1 newly installed, 0 to remove and 39 not upgraded.
Need to get 752 kB of archives.
After this operation, 2,482 kB of additional disk space will be used.
Get:1 http://archive.ubuntu.com/ubuntu bionic-updates/main amd64 sqlite3 amd64 3.22.0-1ubuntu0.4 [752 kB]
Fetched 752 kB in 1s (781 kB/s)
Selecting previously unselected package sqlite3.
(Reading database ... 160772 files and directories currently installed.)
Preparing to unpack .../sqlite3_3.22.0-1ubuntu0.4_amd64.deb ...
Unpacking sqlite3 (3.22.0-1ubuntu0.4) ...
Setting up sqlite3 (3.22.0-1ubuntu0.4) ...
Processing triggers for man-db (2.8.3-2ubuntu0.1) ...


In [ ]:
!sqlite3 --version

3.22.0 2018-01-22 18:45:57 0c55d179733b46d8d0ba4d88e01a25e10677046ee3da1d5b1581e86726f2alt1


As 3.22 is not enough, we're going to install from [source][1]. A good tutorial can be found [here][2]. At the time of this writing, 3.36 is the latest version, however feel free to install any version.
 
[1]: https://www.sqlite.org/download.html
[2]: https://www.vultr.com/docs/how-to-install-the-latest-version-of-sqlite3

In [ ]:
%cd ../../

/content


In [ ]:
!wget https://www.sqlite.org/2021/sqlite-autoconf-3360000.tar.gz; tar xvfz sqlite-autoconf-3360000.tar.gz

--2021-06-20 18:23:06--  https://www.sqlite.org/2021/sqlite-autoconf-3360000.tar.gz
Resolving www.sqlite.org (www.sqlite.org)... 45.33.6.223, 2600:3c00::f03c:91ff:fe96:b959
Connecting to www.sqlite.org (www.sqlite.org)|45.33.6.223|:443... connected.
HTTP request sent, awaiting response... 200 OK
Length: 2977080 (2.8M) [application/x-gzip]
Saving to: ‘sqlite-autoconf-3360000.tar.gz’

sqlite-autoconf-336 100%[===================>]   2.84M  --.-KB/s    in 0.1s    

2021-06-20 18:23:07 (23.2 MB/s) - ‘sqlite-autoconf-3360000.tar.gz’ saved [2977080/2977080]

sqlite-autoconf-3360000/
sqlite-autoconf-3360000/sqlite3.rc
sqlite-autoconf-3360000/README.txt
sqlite-autoconf-3360000/aclocal.m4
sqlite-autoconf-3360000/sqlite3.c
sqlite-autoconf-3360000/Makefile.am
sqlite-autoconf-3360000/configure
sqlite-autoconf-3360000/configure.ac
sqlite-autoconf-3360000/config.guess
sqlite-autoconf-3360000/Makefile.in
sqlite-autoconf-3360000/config.sub
sqlite-autoconf-3360000/install-sh
sqlite-autoconf-3360000/sql

In [ ]:
%cd sqlite-autoconf-3360000/
! ./configure; make; sudo make install

/content/sqlite-autoconf-3360000
checking for a BSD-compatible install... /usr/bin/install -c
checking whether build environment is sane... yes
checking for a thread-safe mkdir -p... /bin/mkdir -p
checking for gawk... no
checking for mawk... mawk
checking whether make sets $(MAKE)... yes
checking whether make supports nested variables... yes
checking for style of include used by make... GNU
checking for gcc... gcc
checking whether the C compiler works... yes
checking for C compiler default output file name... a.out
checking for suffix of executables... 
checking whether we are cross compiling... no
checking for suffix of object files... o
checking whether we are using the GNU C compiler... yes
checking whether gcc accepts -g... yes
checking for gcc option to accept ISO C89... none needed
checking whether gcc understands -c and -o together... yes
checking dependency style of gcc... gcc3
checking for special C compiler options needed for large files... no
checking for _FILE_OFFSET_BITS v

In [ ]:
!cp .libs/libsqlite3.so.0.8.6 /usr/lib/x86_64-linux-gnu/libsqlite3.so.0.8.6

Now verify your version of sqlite.

In [ ]:
!sqlite3 --version

3.36.0 2021-06-18 18:36:39 5c9a6c06871cb9fe42814af9c039eb6da5427a6ec28f187af7ebfb62eafa66e5


Great! Now create the database. 

In [ ]:
%cd ../ImaginaryTicketing/src

/content/ImaginaryTicketing/src


In [ ]:
!sqlite3 utils/database/bot.db < utils/database/bot_schema.sql

Finally, lets run our bot 📈.

In [ ]:
!python3 bot.py

06-20 18:25:19 {bot.py:30} INFO - logging has started
06-20 18:25:19 {client.py:257} WARNING - PyNaCl is not installed, voice will NOT be supported
06-20 18:25:19 {bot.py:85} INFO - Loaded extension 'event'
06-20 18:25:19 {bot.py:85} INFO - Loaded extension 'admin'
06-20 18:25:19 {bot.py:85} INFO - Loaded extension 'smolcmds'
06-20 18:25:19 {bot.py:85} INFO - Loaded extension 'submit'
06-20 18:25:19 {client.py:510} INFO - logging in using static token
06-20 18:25:22 {background.py:111} INFO - aaaa
06-20 18:25:22 {background.py:111} INFO - misc-ticketing-dev-4
06-20 18:25:22 {background.py:111} INFO - ctf-help-0x6f72656f73-0
06-20 18:25:22 {background.py:111} INFO - ctf-help-closed-0x6f72656f73-55
06-20 18:25:22 {background.py:111} INFO - ctf-help-closed-0x6f72656f73-56
06-20 18:25:22 {background.py:111} INFO - ctf-help-0x6f72656f73-58
06-20 18:25:22 {background.py:111} INFO - misc-0x6f72656f73-closed-0
06-20 18:25:22 {background.py:111} INFO - ctf-help-0x6f72656f73-1
06-20 18:25:22 {ba